rpcserver.go

```diff
@@ -1056,7 +1056,12 @@ func (r *rpcServer) sendCoinsOnChain(paymentMap map[string]int64,
 		return nil, err
 	}
 
-	tx, err := r.server.cc.Wallet.SendOutputs(outputs, feeRate, minconf, label)
+	// TODO(halseth): we currently don't check the reserved wallet balance
```

Comment:
So then this PR makes sure you can't fund if you don't have funds on chain, but lets people side step this with `SendCoins`? I guess this would've also existed independent of this specific case given people are still free to use the custom coin selection APIs for this purpose.

Reply:
Alternatively, couldn't we do a dry run here? 

I guess mid-term, we've planned to move coin selection out of the wallet and purely into `lnd` in any case...

Reply:
Good idea, added the dry-run!